src/Aspire.Hosting/Dashboard/DashboardAnnotation.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.Dashboard;
+
+/// <summary>
+/// The annotation that reflects how a resource shows up in the dashboard.
+/// </summary>
+public class DashboardAnnotation(Action<DashboardResourceContext> callback) : IResourceAnnotation
+{
+    /// <summary>
+    /// The context for a dashboard resource.
+    /// </summary>
+    public Action<DashboardResourceContext> Callback { get; } = callback;
+}
+
+/// <summary>
+/// The context for a all of the properties and URLs that should show up in the dashboard for a resource.
+/// </summary>
+public class DashboardResourceContext
+{
+    /// <summary>
+    /// Represents the state of the resource.
+    /// </summary>
+    public string? State { get; set; }
```

Comment:
Is this expected to change over time at all? If so, what would trigger updates? If not, I'm curious what value this adds as a static item, other than populating the column. Are there some examples?

Reply:
I moved to a model where these are immutable snapshots.